code/src/components/app-typography/AppTypography.tsx

```diff
@@ -0,0 +1,64 @@
+import { ReactNode } from "react";
+import { FormattedMessage } from "react-intl";
+import Typography, { TypographyProps } from "@mui/material/Typography";
+import "@/components/app-typography/AppTypography.scss";
+
+type HeadingVariant = "h1" | "h2" | "h3";
+type TextVariant =
+  | "body"
+  | "concept"
+  | "subtitle1"
+  | "subtitle2"
+  | "caption"
+  | "caption-small";
+type AppTypographyVariant = HeadingVariant | TextVariant;
+
+type AppTypographyProps = Omit<TypographyProps, "variant" | "children"> & {
+  variant?: AppTypographyVariant;
+} & (
+    | {
+        translationKey: string;
```

Comment:
There are other props for <FormattedMessage /> that can be useful, especially `values` which gives us a possibility to dynamically insert values in a message. All of them you can find here: https://formatjs.io/docs/react-intl/components/#formattedmessage

Reply:
Fixed